src/app/history/(components)/OrderTable/TableRowOrder.tsx

```diff
@@ -1,35 +1,64 @@
 import { TableCell, TableRow } from "@bleu-fi/ui";
-import { TrashIcon } from "@radix-ui/react-icons";
+import { useRouter } from "next/navigation";
+import { useState } from "react";
 
+import { Checkbox } from "#/components/Checkbox";
 import { TokenInfo } from "#/components/TokenInfo";
-import { StopLossOrderType } from "#/hooks/useOrders";
-import { useRawTxData } from "#/hooks/useRawTxData";
+import { StopLossOrderType } from "#/contexts/ordersContext";
 import { ChainId } from "#/lib/publicClients";
-import { OrderCancelArgs, TRANSACTION_TYPES } from "#/lib/transactionFactory";
 import { IToken } from "#/lib/types";
-import { capitalize, cn, formatDateToLocalDatetime } from "#/utils";
+import { formatDateToLocalDatetime } from "#/utils";
 
-export function TableRowOrder({ order }: { order: StopLossOrderType }) {
-  const { sendTransactions } = useRawTxData();
+import { CancelOrdersDialog } from "../CancelOrdersDialog";
+import { StatusBadge } from "../StatusBadge";
 
-  async function CancelOrder() {
-    const cancelTransactionsData = [
-      {
-        type: TRANSACTION_TYPES.ORDER_CANCEL,
-        hash: order?.hash,
-      } as OrderCancelArgs,
-    ];
 
-    await sendTransactions(cancelTransactionsData);
-  }
-  const disabled = order?.status != "created" && order?.status != "posted";
+interface ITableRowOrder {
+  order: StopLossOrderType;
+  ordersToCancel: string[];
+  setOrdersToCancel: (orders: string[]) => void;
+}
+  
+export function TableRowOrder({
+  order,
+  ordersToCancel,
+  setOrdersToCancel,
+}: ITableRowOrder) {
+  const router = useRouter()
+  const [rowIsSelected, setRowIsSelected] = useState(false);
 
   return (
     <>
-      <TableRow key={order?.id}>
+      <TableRow
+       key={order?.id} 
+       className="border-transparent hover:cursor-pointer hover:bg-brown12"
+       onClick={
+        () => router.push(`/history/order/${order?.hash}`)
+        }
+      >
+
         <TableCell>
           <span className="sr-only"></span>
         </TableCell>
+        <TableCell 
+          onClick={(e) => e.stopPropagation()}
+          className="cursor-default"
+        >
+          <Checkbox
+            id="select-row"
+            onChange={() => {
+              setRowIsSelected(!rowIsSelected);
+              const newOrdersToCancel = !rowIsSelected
+                ? [...ordersToCancel, order.hash]
+                : ordersToCancel.filter(
+                    (orderHash) => orderHash !== order.hash,
+                  );
+              setOrdersToCancel(newOrdersToCancel);
```

Comment:
seems like the logic for row selection and cancelling orders is a bit intertwined here. Do we need both? Shouldn't this component only care about rows being selected or not and the what to do with the selected rows being handled somewhere else?